homeassistant/components/eafm/config_flow.py

```diff
@@ -0,0 +1,54 @@
+"""Config flow to configure flood monitoring gauges."""
+import logging
+
+from aioeafm import get_stations
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+# pylint: disable=unused-import
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class UKFloodsFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a UK Environment Agency flood monitoring config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self):
+        """Handle a UK Floods config flow."""
+        self.stations = {}
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow start."""
+        errors = {}
+
+        if user_input is not None:
+            station = self.stations[user_input["station"]]
+            await self.async_set_unique_id(station, raise_on_progress=False)
+            self._abort_if_unique_id_configured()
+            return self.async_create_entry(
+                title=user_input["station"], data={"station": station},
+            )
+
+        session = async_get_clientsession(hass=self.hass)
+        stations = await get_stations(session)
+
+        self.stations = {}
+        for station in stations:
+            self.stations[station["label"]] = station["stationReference"]
+
+        if not self.stations:
+            return self.async_abort(reason="no_stations")
+
+        return self.async_show_form(
+            step_id="user",
+            errors=errors,
+            data_schema=vol.Schema(
+                {vol.Required("station"): vol.In(sorted(self.stations))}
```

Comment:
Sorry i'm a bit confused here. It's doing what I expect at the moment though - a list of monitoring stations is displayed (by human label) and selecting the human label is correctly mapped back to a station id. What have I missed?

Reply:
It's showing the station reference in the form and returning the station label, right? Is that what we want? Originally we had the inverse.

Reply:
![Screenshot 2020-08-10 at 08 46 12](https://user-images.githubusercontent.com/11544/89761498-69c3e300-dae6-11ea-9da8-b4d38e74c4dc.png)

It's definitely showing the label, then turning the label back into a station reference.

Reply:
Right, it's the `sorted` call that creates a list out of the keys. So the form will return the label, and then we look up the corresponding station reference in the dictionary after receiving the user input on line 31.

It's fine to keep it like this, if the label is ok to use to do the lookup.

Reply:
Yeah the label is fine to use as the lookup. There can be multiple sensors for a label (place/location) but that manifests as multiple measures under the same station, rather than their being duplication station labels. So I think this is fine.